staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -188,19 +152,16 @@ func WithAuditID(ctx context.Context, auditID types.UID) {
 		return
 	}
 	ac := AuditContextFrom(ctx)
-	if ac == nil {
+	if !ac.Enabled() {
```

Comment:
I think storing the `auditID` in the request context is orthogonal to audit logging. A cluster operator can disable audit logging, but we would still want to store the auditID stored in the request context. The auditID is also propagated to the aggregated APIServer(s). 

We have some logs that emit the auditID, for example: 
https://github.com/kubernetes/kubernetes/blob/4db6bde859d912e11ab081a06e8ea17b2e044f5d/staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go#L58
I think we would want to keep the correlation here, even though audit logging is disabled, thoughts?

Reply:
Yeah, that makes sense. You might also be interested in this issue, which has come up a couple of times in sig-apimachinery: https://github.com/kubernetes/kubernetes/issues/101597